### ISSUE TITLE
publish value to specific key name

### ISSUE THREAD

**Issue by qoolq** [ipfs/py-ipfs-api#95]
The IPFS command line utility provides a way to publish an IPFS path for a particular IPNS key by using its name, e.g. `ipfs name publish --key=mykey ...`; ipfsapi doesn't seem to have a way to do this, however.

**Comment by alexander255** [comment]
Thanks for your feedback! We are indeed a bit behind in terms of tracking all the recent API additions atm…

As a temporary workaround you can use the `opts` parameter (for an example see [here](https://github.com/ipfs/py-ipfs-api/issues/84#issuecomment-286453073)); it would be awesome however if you could prepare a tiny commit (see 7c16cc6917080fb631f7e329fca17b16980a5d5e for an example) that adds the parameter in proper. :slightly_smiling_face: 

Let me know if there are any problems!

**Comment by pjz** [comment]
So.. I fixed this, but then wondered if maybe the entire idea of trying to keep up with the go api individual args is wrong.  Why not just do something like:
```
opts = { k.subst('_','-'): kwargs[k] for k in kwargs } 
```
to make `ipfs.name_publish(path, key="mykey")` work as expected?  It's basically taking the `opts` workaround and using it as the main way to handle args.   It does mean that the documentation becomes mostly a pointer to the ipfs-go-api docs, but that might be okay?

**Comment by alexander255** [comment]
I guess technically you're right, but I dislike the idea to be honest: The API isn't that volatile anymore and it's kinda nice to have those parameters represented natively in Python – for one it gives us positional parameters that would break for everybody using them atm. Also `opts` isn't the only "magic parameter" currently supported and we'd basically lock ourselves out of adding more if follow your proposal.

TL;DR: You may be right, but I hope that we can manage with adding the parameters as we go without doing something as drastic. :slightly_smiling_face:
